pages/science/reactors/cvodes.rst

```diff
@@ -0,0 +1,244 @@
+  a single internal step, and is used in ``ReactorNet::step()``.
+
+The result of the ``CVode()`` method is assigned to the ``flag`` object. ``CVode()`` returns 1 or 0, correpsonding to 
+a successful or unsuccessful integration, respectively. 
+
+.. code-block::
+
+    int flag = CVode(m_cvode_mem, tout, m_y, &m_time, CV_NORMAL);
+
+Step 6: ``FuncEval`` class describes ODEs to solve
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+How does ``CVODES`` know what ODE system it should be solving? 
+
+The ODE system was actually already specified using ``CVodeInit()``, one of the methods automatically invoked during the
+``ReactorNet::initialize()`` routine. ``CVODES`` requires that its user provide a C function that defines their ODE, 
+able to compute the right-hand side of the ODE system (dy/dt) for a given value of the independent variable, `t`, 
+and the state vector, ``y``. For more information about ODE right-hand side function requirements, 
+see `CVODES User Guide <https://sundials.readthedocs.io/en/latest/cvodes/Usage/SIM.html#user-supplied-functions>`__.
+
+The ``CVodesIntegrator`` wrapper class provides a useful C++ interface for configuring this C function by pairing with 
+``FuncEval``, an abstract base class for ODE right-hand-side function evaluators (`Documentation 
+<https://cantera.org/documentation/docs-2.6/doxygen/html/d1/dd1/classCantera_1_1FuncEval.html>`__). Classes derived 
+from ``FuncEval`` will implement the evaluation of the provided ODE system.
+
+An ODE right-hand-side evaluator is always needed in the ODE solution process (it's the only way to describe the system!), and for that reason a `FuncEval` object is a required parameter 
+when initializing any type of ``Integrator``.
+
+Let's take a look at how ``ReactorNet`` implements this ``FuncEval`` object. ``ReactorNet`` actually points to itself when 
+defining a ``FuncEval`` type, meaning it defines *itself* as a ``FuncEval`` derivative.
+
+Then, ``ReactorNet`` initializes the ``Integrator``, using a reference to itself (as a ``FuncEval``) from the 
+`this <https://en.cppreference.com/w/cpp/language/this>`__ pointer.
+
+To be a valid ``FuncEval`` object, a ``ReactorNet`` needs to provide implementations for all of ``FuncEval``'s 
+virtual functions, particularly the actual ODE right-hand-side computation 
+function, ``FuncEval::eval()``. Note that this is declared as a `pure virtual 
+<https://en.cppreference.com/w/cpp/language/abstract_class>`__ function, which makes 
+``FuncEval`` an abstract class:
+
+To evaluate the reactor governing equations the following parameters must be known:
+
+@param[in] t time.
+@param[out] LHS pointer to start of vector of left-hand side 
+coefficients for governing equations, length m_nv, default values 1
+@param[out] RHS pointer to start of vector of right-hand side 
+coefficients for governing equations, length m_nv, default values 0
+    
```

Comment:
This needs some formatting work -- it currently just all gets written out as if it was a paragraph.